rohmu/object_storage/google.py

```diff
@@ -402,25 +404,43 @@ def delete_key(self, key: str) -> None:
             self.notifier.object_deleted(key)
 
     def get_contents_to_fileobj(
-        self, key: str, fileobj_to_store_to: BinaryIO, *, progress_callback: ProgressProportionCallbackType = None
+        self,
+        key: str,
+        fileobj_to_store_to: BinaryIO,
+        *,
+        byte_range: Optional[Tuple[int, int]] = None,
+        progress_callback: ProgressProportionCallbackType = None,
     ) -> Metadata:
         path = self.format_key_for_backend(key)
         self.log.debug("Starting to fetch the contents of: %r to %r", path, fileobj_to_store_to)
         next_prog_report = 0.0
         last_log_output = 0.0
+        self._validate_byte_range(byte_range)
         with self._object_client(not_found=path) as clob:
             metadata, obj_size = self._metadata_for_key(clob, path)
-            reporter = Reporter(StorageOperation.get_file, size=min(obj_size, DOWNLOAD_CHUNK_SIZE))
+            if byte_range is None:
+                size_to_download = obj_size
+            else:
+                size_to_download = min(
+                    obj_size - byte_range[0], byte_range[1] - byte_range[0] + 1 if byte_range else obj_size
```

Comment:
`if byte_range` here looks spurious?

Reply:
Yes, I guess originally I was handling everything in a single expression and I forgot to remove that condition.